### PR TITLE
Modified API and media fields to enable video preview on link creation.

### PR DESCRIPTION
**Pull request by zedee**
I have modified the Media controller API in order to be able to display a preview of a video if it matches the most popular mimeTypes, fixing the issue that would render a broken 'img' (JSON was returning an empty path as well if a video was linked, because the way of getting the thumbnail via Imagy).

My version only differs on lines 74 and 78, perhaps I have not pulled from the correct way in my fork or has it been added after? Because I did not have them in the source.

### CODE REVIEW

File: Http/Controllers/Api/MediaController.php [hunk 1]
```diff
@@ -37,74 +45,100 @@ public function __construct(FileService $fileService, FileRepository $file, Imag
...
+     * @param Request $request            
      */
     public function linkMedia(Request $request)
     {
         $mediaId = $request->get('mediaId');
         $entityClass = $request->get('entityClass');
         $entityId = $request->get('entityId');
-        $order = $request->get('order');
-
+
```
Comment (mikemand) [inline Http/Controllers/Api/MediaController.php]: Where did the `order` go? This is important for the multiple media selections.

Comment (zedee) [inline Http/Controllers/Api/MediaController.php]: Sorry, my bad. I made a CTRL+A +V from the wrong source code.
I put again those lines in place. I commited & pushed it.
Do I need to make again the pull request?

Comment (mikemand) [inline Http/Controllers/Api/MediaController.php]: Thank you @zedee. :)

File: Http/Controllers/Api/MediaController.php [hunk 2]
```diff
@@ -37,74 +45,102 @@ public function __construct(FileService $fileService, FileRepository $file, Imag
...
+        $imageable = DB::table('media__imageables')->whereFileId($mediaId)
+            ->whereZone($zone)
+            ->whereImageableType($entityClass)
+            ->first();
         $file = $this->file->find($imageable->file_id);
-
-        $thumbnailPath = $this->imagy->getThumbnail($file->path, 'mediumThumb');
-
+        
+        if ($file->mimetype == 'video/mp4' || $file->mimetype == 'video/ogv' || $file->mimetype == 'video/webm') {
```
Comment (mikemand) [inline Http/Controllers/Api/MediaController.php]: Just a small nitpick (you do not need to make this change if you don't want to), I would use something like `strpos($file->mimetype, 'video') !== 0` so you catch all possible `video/*` mimetypes. It also saves you a bit of typing. :)

Comment (zedee) [inline Http/Controllers/Api/MediaController.php]: Oh, that's a fantastic idea. Limiting mimeTypes to popular formats would be restrictive. Your suggestion is undoubtely applied, thanks! :+1:

**Comment by mikemand** [comment]
This is excellent. :) *poke @nWidart*

**Comment by nWidart** [comment]
Thank you very much for the contribution @zedee 

I will look at this as soon as possible! 😄

**Comment by zedee** [comment]
You welcome! Thanks for the suggestions and the feedback :smile:

**Comment by nWidart** [comment]
I haven't forgotten this, just lots to do :(

**Comment by nWidart** [comment]
Thank you for the contribution. :)

However the new way of adding media via the partial has drastically changed. You might have to go over that once I add it on the Media module here. (It's currently on the repository for a specific client).

**Comment by nWidart** [comment]
You can preview the new changes here : https://github.com/AsgardCms/Platform/issues/195#issuecomment-219048218

**Pull request merged**